lib/util/visit-require.js

```diff
@@ -10,28 +10,33 @@ const {
     ReferenceTracker,
     getStringIfConstant,
 } = require("@eslint-community/eslint-utils")
-const isCoreModule = require("is-core-module")
+const isBuiltinModule = require("is-builtin-module")
 const getResolvePaths = require("./get-resolve-paths")
 const getTryExtensions = require("./get-try-extensions")
 const ImportTarget = require("./import-target")
 const stripImportPathParams = require("./strip-import-path-params")
 
+/**
+ * @typedef {Object} VisitRequireOptions
```

Comment:
```suggestion
 * @typedef VisitRequireOptions
```

When using `@property` no type is needed in `@typedef`, its implied that its an object